vm-ubuntu-2204-amd64-uefi/gh-actions-overlay.nix

```diff
@@ -0,0 +1,113 @@
+{
+  pkgs ? import <nixpkgs> {},
+  system ? builtins.currentSystem,
+  image ? pkgs.callPackage ./default.nix {},
+  ...
+}: let
+  inherit (pkgs) callPackage dasel;
+  githubRunner = import ./github-runner.nix {inherit pkgs;};
+  overlayedImage = pkgs.vmTools.runInLinuxVM (
+    pkgs.runCommand "nixos-sun-baseline-image"
+    {
+      memSize = 768;
+      preVM = ''
+        mkdir -p $out
+
+        # Locate the image manifest within the store expression:
+        IMAGE_HASH="$(cat "${image}/image.txt")"
+        MANIFEST_PATH="${image}/images/''${IMAGE_HASH:0:2}/''${IMAGE_HASH:2:2}/''${IMAGE_HASH:4:2}/$IMAGE_HASH"
+
+        # Extract the name of the "head" blob in the image:
+        HEAD_BLOB="$(${dasel}/bin/dasel -f "$MANIFEST_PATH" -r toml -w - '.org\.tockos\.treadmill\.manifest-ext\.base\.attrs.org\.tockos\.treadmill\.image\.qemu_layered_v0\.head')"
+
+        # Retrieve the head blob's SHA-256 digest:
+        BLOB_HASH="$(${dasel}/bin/dasel -f "$MANIFEST_PATH" -r toml -w - '.org\.tockos\.treadmill\.manifest-ext\.base\.blobs.'"$HEAD_BLOB"'.org\.tockos\.treadmill\.manifest-ext\.base\.sha256-digest')"
+        BLOB_PATH="${image}/blobs/''${BLOB_HASH:0:2}/''${BLOB_HASH:2:2}/''${BLOB_HASH:4:2}/$BLOB_HASH"
+
+        # The below variable is magically picked up and exposed to the VM.
+        diskImage=disk.qcow2
+        echo "Creating $diskImage based on $BLOB_PATH"
+        ${pkgs.qemu}/bin/qemu-img create -b "$BLOB_PATH" -F qcow2 -f qcow2 "$diskImage"
+      '';
+      postVM = ''
+        mkdir -p $out
+        cp $diskImage $out/overlay.qcow2
+      '';
+      buildInputs = [];
+    }
+    ''
+      mkdir -p /mnt
+      ${pkgs.mount}/bin/mount /dev/vda2 /mnt
+      ls /mnt
+
+      cp -r ${githubRunner} /mnt/opt/actions-runner
+
+      cat > "/mnt/etc/systemd/system/github-runner.service" <<EOF
+      [Unit]
+      Description=GitHub Actions Runner
+      After=network.target
+
+      [Service]
+      ExecStartPre=/bin/bash -c 'export REPO_URL=\$(cat /run/tml/repo-url) && export TOKEN=\$(cat /run/tml/ghrunner-token) && export JOB_ID=\$(cat /run/tml/job-id) && opt/actions-runner/config.sh --url \$REPO_URL --token \$TOKEN --name tml-ghactionsrunner-\$JOB_ID --labels tml-ghactionsrunner-\$JOB_ID --unattended --ephemeral'
+      ExecStart=/opt/actions-runner/run.sh
+      Restart=on-failure
+      User=root
+      WorkingDirectory=/opt/actions-runner
+
+      [Install]
+      WantedBy=multi-user.target
+      EOF
+    ''
+  );
+in
+  pkgs.stdenv.mkDerivation {
+    pname = "new-image-store";
+    version = "1.0";
+    buildInputs = [image overlayedImage];
+    src = ./.;
+    buildPhase = ''
+      cp ${overlayedImage}/overlay.qcow2 $TMPDIR/overlay.qcow2
```

Comment:
Why do we copy here right now? The file isn't modified anywhere?

I do think that we need to modify this file, but only to edit the image's backing path into a relative one that'll work given the image store layout. So this file will need to have its backing file path be updated to something like `../../../xx/xx/xx/<backing file hash>`.

It'd be great if we could add this.